### ISSUE TITLE
Install failure on arch linux

### ISSUE THREAD

**Issue by stanford-scs** [OpenZWave/python-openzwave#126]
python-openzwave fails to install on arch linux, due to the following installation problem:

```  
  b'      snprintf(usercode, sizeof(usercode), "%s %d", usercode, (int)_data[12+i]);\n'
  b'      snprintf(usercode, sizeof(usercode), "%s %d", usercode, (int)_data[12+i]);\n'
  
  b'               ^~~~~~~~                             ~~~~~~~~\n'
  b'               ^~~~~~~~                             ~~~~~~~~\n'
  
  b'cc1plus: all warnings being treated as errors\n'
  b'cc1plus: all warnings being treated as errors\n'
  
  b'make[1]: *** [/tmp/pip-install-pbfhpvd1/python-openzwave/openzwave-embed/open-zwave-master/cpp/build/support.mk:139: /tmp/pip-install-pbfhpvd1/python-openzwave/openzwave-embed/open-zwave-master/.lib/DoorLockLogging.o] Error 1\n'
  b'make[1]: *** [/tmp/pip-install-pbfhpvd1/python-openzwave/openzwave-embed/open-zwave-master/cpp/build/support.mk:139: /tmp/pip-install-pbfhpvd1/python-openzwave/openzwave-embed/open-zwave-master/.lib/DoorLockLogging.o] Error 1\n'
...
  g++: error: openzwave-embed/open-zwave-master/libopenzwave.a: No such file or directory
 ```

This is obviously related to [this bug](https://github.com/OpenZWave/open-zwave/pull/1513), but I get a similar problem even when I try installing a manually patched and compiled libopenzwave.a in /usr/lib and using the non-embedded variants of the python package.

Could you please either fix this problem, or even better (in case future compilation problems come up) explain in the instructions how to pull the embedded open-zwave code from a different repository/branch or how to get the non-embedded versions to use a library already installed in /usr/lib?

**Comment by bibi21000** [comment]
> This is obviously related to this bug, but I get a similar problem even when I try installing a manually patched and compiled libopenzwave.a in /usr/lib and using the non-embedded variants of the python package.

Which flavor of python-openzwave do you use ?

> Could you please either fix this problem, or even better (in case future compilation problems come up) explain in the instructions how to pull the embedded open-zwave code from a different repository/branch or how to get the non-embedded versions to use a library already installed in /usr/lib?

Your problem should be solve using the right flavor. Could you provide the way you compile / install openzwave and python-openzwave, or better some copy/paste ?

**Comment by stanford-scs** [comment]
I've tried several.  The original bug report was a transcript from the default (embed) flavor.  I've attached a log from the shared flavor to this message.  The problem in this build seems to be that arch installs the headers in a place where python doesn't look for them.  I think the only reliable way to build this in multiple environments would be to allow the "embed" flavor to pull from a git repository with the necessary patches to compile, because then the python build process would know exactly where everything is.  If there's some way to do that, please put in in the instructions.  If not, please consider adding it rather than hard-coding the open-zwave repository.

[typescript.log](https://github.com/OpenZWave/python-openzwave/files/2217319/typescript.log)

**Comment by bibi21000** [comment]
python-openzwave use the classical pkg-config tools to locate libs,
headers, .... si it should work on every sysyem using it

Look at the result of :

pkg-config libopenzwave --cflags --libs

If pkg-config infos are not available, check your openzwave installation.

2018-07-22 21:41 GMT+02:00 stanford-scs <notifications@github.com>:

**Comment by stanford-scs** [comment]
```
$ pkg-config libopenzwave --cflags --libs
-I/usr/include/openzwave/ -lopenzwave 
```

And the files are under `/usr/include/openzwave` as shown in my typescript.

**Comment by bibi21000** [comment]
Should be fixed in 0.4.7
Thanks
